src/builder/pdb.rs

```diff
@@ -0,0 +1,268 @@
+    /// We use u16, as [`IntOrString`] takes an i32 and we don't want to allow negative numbers. u16 will always fit in i32.
+    min_available: Option<u16>,
+    /// Tracks wether either `max_unavailable` or `min_available` are set
+    _constraints: Constraints,
+}
+
+impl PdbBuilder<(), (), ()> {
+    pub fn new() -> Self {
+        PdbBuilder::default()
+    }
+
+    pub fn new_for_role<T: Resource<DynamicType = ()>>(
+        owner: &T,
+        app_name: &str,
+        role: &str,
+        operator_name: &str,
+        controller_name: &str,
+    ) -> OperatorResult<PdbBuilder<ObjectMeta, LabelSelector, ()>> {
+        let role_selector_labels = role_selector_labels(owner, app_name, role);
+        let metadata = ObjectMetaBuilder::new()
+            .namespace_opt(owner.namespace())
+            .name(format!("{}-{}", owner.name_any(), role))
+            .ownerreference_from_resource(owner, None, Some(true))?
+            .with_labels(role_selector_labels.clone())
+            .with_label(
+                APP_MANAGED_BY_LABEL.to_string(),
+                format_full_controller_name(operator_name, controller_name),
+            )
+            .build();
+        Ok(PdbBuilder {
+            metadata,
+            selector: LabelSelector {
+                match_expressions: None,
+                match_labels: Some(role_selector_labels),
+            },
+            ..PdbBuilder::default()
+        })
+    }
+
+    pub fn metadata(self, metadata: impl Into<ObjectMeta>) -> PdbBuilder<ObjectMeta, (), ()> {
+        PdbBuilder {
+            metadata: metadata.into(),
+            selector: self.selector,
+            max_unavailable: self.max_unavailable,
+            min_available: self.min_available,
+            _constraints: self._constraints,
+        }
+    }
+}
+
+impl PdbBuilder<ObjectMeta, (), ()> {
+    pub fn selector(self, selector: LabelSelector) -> PdbBuilder<ObjectMeta, LabelSelector, ()> {
+        PdbBuilder {
+            metadata: self.metadata,
+            selector,
+            max_unavailable: self.max_unavailable,
+            min_available: self.min_available,
+            _constraints: self._constraints,
+        }
+    }
+}
+
+impl PdbBuilder<ObjectMeta, LabelSelector, ()> {
+    pub fn max_unavailable(
+        self,
+        max_unavailable: u16,
+    ) -> PdbBuilder<ObjectMeta, LabelSelector, bool> {
+        PdbBuilder {
+            metadata: self.metadata,
+            selector: self.selector,
+            max_unavailable: Some(max_unavailable),
+            min_available: self.min_available,
+            _constraints: true, // Some dummy value to set Constraints to something other than ()
+        }
+    }
+
+    #[deprecated(
+        since = "0.51.0",
+        note = "It is strongly recommended to use [`max_unavailable`]. Please read the ADR on Pod disruptions before using this function."
+    )]
```

Comment:
I'm not sure if we should immediately deprecate this function. We might want to add some doc comment instead.

Reply:
Have spoken about this offline and came to the conclusion we should try to prevent us from using minAvailable.
We could rename it to `with_min_available_i_know_what_im_doing`, but that does not sound ideal either